static/app/components/modals/sudoModal.tsx

```diff
@@ -84,29 +82,43 @@ function SudoModal({
     superuserReason,
   } = state;
 
-  const {loadOrganization} = useContext(OrganizationLoaderContext) || {};
-  const router = useRouter();
-  const api = useApi();
+  const {organizationPromise} = useContext(OrganizationLoaderContext);
   const location = useLocation();
 
   useEffect(() => {
-    const getAuthenticators = () => {
-      if (!loadOrganization) return;
+    const getAuthenticators = async () => {
       try {
-        loadOrganization().finally(async () => {
-          const fetchedAuthenticators = await api.requestPromise('/authenticators/');
-          setState(prevState => ({
-            ...prevState,
-            authenticators: fetchedAuthenticators ?? [],
-          }));
-        });
+        // Await all preload requests
+        await Promise.allSettled([
+          organizationPromise,
+          ...Object.values(window.__sentry_preload),
+        ]);
       } catch {
         // ignore errors
       }
+
+      // Fetch authenticators after preload requests to avoid overwriting session cookie
+      try {
+        const fetchedAuthenticators = await api.requestPromise('/authenticators/');
+        setState(prevState => ({
+          ...prevState,
+          authenticators: fetchedAuthenticators ?? [],
+          isLoading: false,
+        }));
+      } catch {
+        setState(prevState => ({
+          ...prevState,
+          isLoading: false,
+        }));
+      }
     };
 
+    if (!organizationPromise) {
+      return;
+    }
```

Comment:
I believe we can delete this, we only added it previously b/c we were calling `.finally` on it.

Reply:
looks like this is being used to gate when we call `getAuthenticators` is invoked and we need it because the initial value is set to null.